iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/unique.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <utility>
+
+namespace iox
+{
+template <typename T>
+class unique
+{
+  public:
+    struct invalid_t
+    {
+        explicit constexpr invalid_t()
+        {
+        }
+    };
+
+    static constexpr invalid_t invalid{};
```

Comment:
I recommend moving invalid outside of unique. Then you could use it more easily. Here take a look
```
unique<FatClassName> a(unique<FatClassName>::invalid);
// or better version
unique<FatClassName> a(invalid);
```

Since you are reimplementing an optional here. ```invalid``` corresponds to std::nullopt and the STL uses this approach